File: woori_link/woori_link_flutter/lib/api/risk_api.dart
import 'dart:convert';
import 'package:http/http.dart' as http;
import '../constants.dart';
import 'http_client.dart';

class RiskApi {
  static Future<Map<String, dynamic>?> getLatestRisk(int seniorId) async {
    final res = await http.get(
      Uri.parse('$baseUrl/risk/senior/$seniorId/latest'),
      headers: await authHeaders(),
    );
    if (res.statusCode == 200) return jsonDecode(res.body);
    return null;
  }

  static Future<Map<String, dynamic>> assessRisk(int seniorId) async {
    final res = await http.post(
      Uri.parse('$baseUrl/risk/assess/$seniorId'),
      headers: await authHeaders(),
    );
    if (res.statusCode == 200) return jsonDecode(res.body);
    throw Exception('위험도 평가 실패');
  }
}

File: woori_link/woori_link_react/src/api/riskApi.js
import api from './index'

export const getLatestRisk = (seniorId) => api.get(`/risk/senior/${seniorId}/latest`)
export const getHighRisk = () => api.get('/risk/high-risk')
export const assessRisk = (seniorId) => api.post(`/risk/assess/${seniorId}`)
export const assessAll = () => api.post('/risk/assess-all')
